sync-test/src/support.rs

```diff
@@ -0,0 +1,309 @@
+/* Any copyright is dedicated to the Public Domain.
+   http://creativecommons.org/publicdomain/zero/1.0/ */
+
+// This is required to prevent warnings about unused functions in this file just
+// because it's unused in a single file (tests that don't use every function in
+// this module will get warnings otherwise).
+#![allow(dead_code)]
+
+use fxa_client::{self, FirefoxAccount, Config as FxaConfig};
+use logins_sql::{Login, PasswordEngine};
+use logins_sql::Result as LoginResult;
+
+use url::Url;
+
+use std::env;
+use std::path::PathBuf;
+use std::collections::HashMap;
+use std::sync::{Once, ONCE_INIT, Arc};
+use failure;
+use serde_json;
+use sync15_adapter::{Sync15StorageClientInit, KeyBundle};
+use env_logger;
+
+type FailureResult<T> = Result<T, failure::Error>;
+
+pub const CLIENT_ID: &str = "98adfa37698f255b"; // Hrm...
+pub const SYNC_SCOPE: &str = "https://identity.mozilla.com/apps/oldsync";
+
+// TODO: This is wrong for dev?
+pub const REDIRECT_URI: &str = "https://lockbox.firefox.com/fxa/ios-redirect.html";
+
+lazy_static! {
+    // Figures out where `sync-test/helper` lives. This is pretty gross, but once
+    // https://github.com/rust-lang/cargo/issues/2841 is resolved it should be simpler.
+    // That said, it's possible we should probably just rewrite that script in rust instead :p.
+    static ref HELPER_SCRIPT_DIR: PathBuf = {
+        let mut path = env::current_exe().expect("Failed to get current exe path...");
+        // Find `target` which should contain this program.
+        while path.file_name().expect("Failed to find target!") != "target" {
+            path.pop();
+        }
+        // And go up once more, to the root of the workspace.
+        path.pop();
+        // TODO: it would be nice not to hardcode these given that we're
+        // planning on moving stuff around, but such is life.
+        path.push("sync-test");
+        path.push("helper");
+        path
+    };
+}
+
+fn run_helper_command(cmd: &str, cmd_args: &[&str]) -> Result<String, failure::Error> {
+    use std::process::{self, Command};
+    // This `Once` is used to run `npm install` first time through.
+    static HELPER_SETUP: Once = ONCE_INIT;
+    HELPER_SETUP.call_once(|| {
+        let dir = &*HELPER_SCRIPT_DIR;
+        env::set_current_dir(dir).expect("Failed to change directory...");
+
+        // Let users know why this is happening even if `log` isn't enabled.
+        println!("Running `npm install` in `integration-test-helper` to ensure it's usable");
+
+        let mut child = Command::new("npm")
```

Comment:
Can we take `npm` and `node` from `$NPM` and `$NODE`, falling back to `"npm"` and `"node"`?  (Let's ease our automation configuration up front.)